daemon/src/db.rs

```diff
@@ -323,22 +340,204 @@ impl Connection {
+
+        Ok(())
+    }
+
+    async fn load_cfd<C>(&self, id: OrderId, args: C::CtorArgs) -> Result<C>
+    where
+        C: ClosedCfdAggregate,
+    {
+        if self.is_closed_cfd(id).await? {
+            let closed_cfd = self.load_closed_cfd(id, args).await?;
+
+            return Ok(closed_cfd);
+        }
+
+        let cfd = self.load_open_cfd(id, args).await?;
+
+        Ok(cfd)
+    }
+
+    /// Load a closed CFD from the database.
+    async fn load_closed_cfd<C>(&self, id: OrderId, args: C::CtorArgs) -> Result<C>
+    where
+        C: ClosedCfdAggregate,
+    {
+        let mut conn = self.inner.acquire().await?;
+
+        let cfd = sqlx::query!(
+            r#"
+            SELECT
+                uuid as "uuid: model::OrderId",
+                position as "position: model::Position",
+                initial_price as "initial_price: model::Price",
+                taker_leverage as "taker_leverage: model::Leverage",
+                n_contracts as "n_contracts: model::Contracts",
+                counterparty_network_identity as "counterparty_network_identity: model::Identity",
+                role as "role: model::Role",
+                fees as "fees: model::Fees",
+                expiry_timestamp,
+                lock_txid as "lock_txid: model::Txid",
+                lock_dlc_vout as "lock_dlc_vout: model::Vout"
+            FROM
+                closed_cfds
+            WHERE
+                closed_cfds.uuid = $1
+            "#,
+            id
+        )
+        .fetch_one(&mut conn)
+        .await?;
+
+        let expiry_timestamp = OffsetDateTime::from_unix_timestamp(cfd.expiry_timestamp)?;
+
+        let collaborative_settlement = load_collaborative_settlement_tx(&mut conn, id).await?;
+
+        let commit = load_commit_tx(&mut conn, id).await?;
+
+        let non_collaborative_settlement = load_cet(&mut conn, id).await?;
+
+        let refund = load_refund_tx(&mut conn, id).await?;
+
+        let cfd = ClosedCfd {
+            id,
+            position: cfd.position,
+            initial_price: cfd.initial_price,
+            taker_leverage: cfd.taker_leverage,
+            n_contracts: cfd.n_contracts,
+            counterparty_network_identity: cfd.counterparty_network_identity,
+            role: cfd.role,
+            fees: cfd.fees,
+            expiry_timestamp,
+            lock: Lock {
+                txid: cfd.lock_txid,
+                dlc_vout: cfd.lock_dlc_vout,
+            },
+            collaborative_settlement,
+            commit,
+            non_collaborative_settlement,
+            refund,
+        };
+
+        Ok(C::new_closed(args, cfd))
+    }
+
+    /// Check if the CFD corresponding to the `id` given is in the
+    /// `closed_cfds` table.
+    async fn is_closed_cfd(&self, id: OrderId) -> Result<bool> {
+        let mut conn = self.inner.acquire().await?;
+
+        let cfd = sqlx::query!(
+            r#"
+            SELECT
+                uuid as "id: model::OrderId"
```

Comment:
Given you don't care about the `uuid` you could just `select 1` which is a bit more performant than selecting the actual item. 

```suggestion
1
```

Reply:
Thanks for the suggestion! I think this doesn't work because `sqlx` expects a Rust identifier here i.e. it won't compile :(

Reply:
stupid sqlx ...